### PR TITLE
[FIX](complextype)fix shrink in topN for complex type

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by eldenmoon — APPROVED** [review]
LGTM

File: be/src/exec/rowid_fetcher.cpp [hunk 1]
```diff
@@ -193,6 +193,24 @@ Status RowIDFetcher::_merge_rpc_results(const PMultiGetRequest& request,
     return Status::OK();
 }
 
+bool _has_char_type(const TypeDescriptor& desc) {
+    switch (desc.type) {
+    case TYPE_CHAR:
+        return true;
+    case TYPE_ARRAY:
+    case TYPE_MAP:
+    case TYPE_STRUCT:
```
Comment (xiaokang) [inline be/src/exec/rowid_fetcher.cpp:202]: can we make it more general to just check children

Comment (yiguolei) [reply be/src/exec/rowid_fetcher.cpp:202]: could not. It has to check the child recursively.....

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Pull request merged**